Packs/FortiGate/Integrations/FortiGate/FortiGate.py

```diff
@@ -103,1209 +2152,3745 @@ def create_addr_string(list_of_addr_data_dicts):
+        raw_response=response,
+    )
+
+
+@logger
+def update_firewall_service_command(client: Client, args: dict[str, Any]) -> CommandResults:
+    """Update firewall services.
+
+    Args:
+        client (Client): Session to Fortigate to run API requests.
+        args (dict[str, Any]): Arguments passed down by the CLI to configure the request.
+
+    Raises:
+        DemistoException: If the input protocol type is different from the API.
+        DemistoException: If `action` was given without arguments TCP/UDP/SCTP parameters.
+
+    Returns:
+        CommandResults: Outputs of the command that represent an entry in warroom.
+    """
+    name = args.get("name", "")
+    vdom = args.get("vdom", DEFAULT_VDOM)
+    comment = args.get("comment")
+    category = args.get("category")
+    start_ip = args.get("start_ip")
+    end_ip = args.get("end_ip")
+    fqdn = args.get("fqdn")
+    tcp_port_ranges = argToList(args.get("tcpRange"))
+    udp_port_ranges = argToList(args.get("udpRange"))
+    sctp_port_ranges = argToList(args.get("sctpRange"))
+    icmp_type = arg_to_number(args.get("icmp_type"))
+    icmp_code = arg_to_number(args.get("icmp_code"))
+    ip_protocol = arg_to_number(args.get("ip_protocol"))
+    action = args.get("action")
+
+    if bool(action) != any((tcp_port_ranges, udp_port_ranges, sctp_port_ranges)):
+        raise DemistoException(f"'action' and '{TCP_UDP_SCTP}' must be set together.")
+
+    validate_optional_ipv4_addresses(start_ip, end_ip)
+
+    input_protocol_type = None
+    response = client.list_firewall_services(name, vdom)
+    result = extract_first_result(response)
+
+    if any(
+        (
+            start_ip,
+            end_ip,
+            fqdn,
+            tcp_port_ranges,
+            udp_port_ranges,
+            sctp_port_ranges,
+            args.get("icmp_version"),
+            icmp_type,
+            icmp_code,
+            ip_protocol,
+        )
+    ):
+        input_protocol_type = get_service_type(args)
+        api_protocol_type = result.get("protocol")
+
+        if input_protocol_type != api_protocol_type:
+            raise DemistoException(
+                f"The service '{name}' is of type '{api_protocol_type}',"
+                f" which is not compatible with the requested type '{input_protocol_type}'."
+            )
+
+    if action:
+        port_ranges = handle_action_for_port_ranges(
+            obj=result,
+            action=action,
+            tcp_port_ranges=tcp_port_ranges,
+            udp_port_ranges=udp_port_ranges,
+            sctp_port_ranges=sctp_port_ranges,
+        )
+
+    response = client.update_firewall_service(
+        name=name,
+        vdom=vdom,
+        comment=comment,
+        category=category,
+        protocol_type=input_protocol_type,
+        start_ip=start_ip,
+        end_ip=end_ip,
+        fqdn=fqdn,
+        icmp_type=icmp_type,
+        icmp_code=icmp_code,
+        ip_protocol=ip_protocol,
+        **port_ranges,
```

Comment:
Call the arguments explicitly. port_ranges.get('..')

Reply:
updated.